Code/Tools/ProjectManager/Source/ProjectManagerSettings.cpp

```diff
@@ -14,46 +14,235 @@
 
 namespace O3DE::ProjectManager
 {
-    void SaveProjectManagerSettings()
+    namespace PMSettings
     {
-        auto settingsRegistry = AZ::SettingsRegistry::Get();
-        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
-        dumperSettings.m_prettifyOutput = true;
-        dumperSettings.m_jsonPointerPrefix = ProjectManagerKeyPrefix;
+        bool SaveProjectManagerSettings()
+        {
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
```

Comment:
A better design might be to make a ProjectSettingsInterface & ProjectSettings implementation that is owned by the Application which, when created, stores the settingsregistry in an m_settingsRegistry variable and handles a missing settings registry there immediately.  Then in all these other functions you don't have to get/check the settings registry.

Calls like
```PMSettings::SetProjectBuiltSuccessfully(m_projectInfo, true);```
would become
```ProjectSettingsInterface::Get()->SetProjectBuiltSuccessfully(m_projectInfo, true);```

thoughts?

Reply:
Sounds awesome! Was considering making it a singleton but may as well go all the way with how complex this PR ended up being